### ISSUE TITLE
Head title callback not called when <Head> title property is empty

### ISSUE THREAD

**Issue by Propaganistas** [inertiajs/inertia#938]
### Versions:

- `@inertiajs/inertia` version: 0.10.0
- `@inertiajs/inertia-vue3` version: 0.5.1

### Describe the problem:

The docs put forward two ways for managing the formatting of a meta title: use the `title` callback when creating the Inertia app and use the shipped `Head` component, or create a custom component extending the `Head` component itself.

Both are announced to be able to do exactly the same thing: accepting a desired title and modifying it to, for example, append the application's name, or provide a default.

However the `title` callback method seems to lack a specific case: when `<Head>` receives an empty-like (`""`, `null`, `undefined`, `false`) value the callback never gets called, the (potentially defined) default is not set and the meta title is omitted.

### Steps to reproduce:

```js
createInertiaApp({
    title: title => title ? `${title} - My App` : 'My App'
  // ...
})
```

```js
<template>
    <Head :title="null" />
</template>
```

**Comment by claudiodekker** [comment]
Hey,

So, technically, the `null` title isn't supported, as that's not a string, but you're right that it makes sense to have a default title when a title callback is defined, which currently isn't working. This will be fixed in #1055

**Comment by soheilghs** [comment]
this doesn't work for me yet
